pkg/tbtc/deposit_sweep.go

```diff
@@ -47,6 +54,26 @@ const (
 	depositSweepBroadcastCheckDelay = 1 * time.Minute
 )
 
+// DepositSweepProposal represents a deposit sweep proposal issued by a
+// wallet's coordination leader.
+type DepositSweepProposal struct {
+	WalletPublicKeyHash [20]byte
+	DepositsKeys        []struct {
+		FundingTxHash      bitcoin.Hash
+		FundingOutputIndex uint32
+	}
+	SweepTxFee           *big.Int
+	DepositsRevealBlocks []*big.Int
+}
+
+func (dsp *DepositSweepProposal) actionType() WalletActionType {
+	return ActionDepositSweep
+}
+
+func (dsp *DepositSweepProposal) validityBlocks() uint64 {
+	return depositSweepProposalValidityBlocks
+}
```

Comment:
This is something that should be proposed by the leader and validated by all followers to double-check they operate on the same value before the protocol execution starts.

Reply:
I can't agree. The design proposed in RFC 12 assumes the leader does not propose action execution timing. The only responsibility of the leader is assembling an action proposal. The execution start block is always the last block of the window and the proposal's validity (so, execution timeout) is fixed. The less the leader is supposed to propose, the better. This reduces complexity and room for manipulation. Do you see any particular reason to let the leader propose the validity?

Reply:
My comment was not clear enough. If `validityBlocks` is a function of `DepositSweepProposal`, one can assume - and that was my assumption as well - that the validity blocks will be eventually sent over the wire like the rest of the proposal fields and that the receiver will check if it is in some expected range. This could in theory give some better flexibility for upgrades and let to adjust the time based on the current situation in the network. However, I don't think this is necessary at this stage. It feels like an additional complexity that we may never need. The version you described is better, but then I would remove the `validityBlocks()` from `DepositSweepProposal` and had it defined as a constant somewhere else. The same for `validityBlocks()` for all other proposals.

Reply:
>  I don't think that's a big problem if we don't include it in the serialized proposal version sent over the wire.

If we don't do that, let's just make it crystal-clear in the docs for `validityBlocks()` to avoid confusion and don't let the future-us have a great idea of making it a marshaled/unmarshalled field.